cli/tools/test/mod.rs

```diff
@@ -561,22 +598,34 @@ pub async fn run_tests_for_worker(
     }
     sender.send(TestEvent::Wait(desc.id))?;
 
-    // TODO(bartlomieju): this is a nasty (beautiful) hack, that was required
-    // when switching `JsRuntime` from `FuturesUnordered` to `JoinSet`. With
-    // `JoinSet` all pending ops are immediately polled and that caused a problem
-    // when some async ops were fired and canceled before running tests (giving
-    // false positives in the ops sanitizer). We should probably rewrite sanitizers
-    // to be done in Rust instead of in JS (40_testing.js).
+    // Poll event loop once, to allow all ops that are already resolved, but haven't
+    // responded to settle.
+    // TODO(mmastrac): we should provide an API to poll the event loop until no futher
+    // progress is made.
```

Comment:
This comment is a bit unclear to me - how would we figure out that "no further progress" is made?

Reply:
It would be something like an option to the poll method -- if the option is set we would return "ready" if the event loop did not make progress on ops, modules, timers or anything else (ie dispatch).

We currently return pending whether we did work or not and this would allow a caller to determine when the runtime has "settled" into a steady state.

This probably requires some research.

Reply:
Should I go ahead with the refactor we discussed in the past, where `poll_event_loop` would return an enum describing state of the event loop instead of `Poll::Pending/Poll::Ready`?